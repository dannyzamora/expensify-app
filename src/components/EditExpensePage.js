import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startEditExpenses, startRemoveExpenses } from '../actions/expenses';

export class EditExpensePage extends React.Component {
  onSubmit = (expense) => {
    this.props.startEditExpenses(this.props.expense.id, expense);
    this.props.history.push('/');
  }
  onRemove = () => {
    this.props.startRemoveExpenses({ id: this.props.expense.id });
    this.props.history.push('/');
  }
  render() {
    return (

      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Edit Expense</h1>
          </div>

        </div>

        <div className="content-container">
          <ExpenseForm
            expense={this.props.expense}
            onSubmit={this.onSubmit}
          />
          <div>
            <button className="button button--secondary" onClick={this.onRemove}>Remove Expense</button>
          </div>

        </div>

      </div>


    );

  }
}


const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
  };
};

const mapDispatchToProps = (dispatch) => (
  {
    startEditExpenses: (id, expense) => dispatch(startEditExpenses(id, expense)),
    startRemoveExpenses: (data) => dispatch(startRemoveExpenses(data))
  }
)

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);
