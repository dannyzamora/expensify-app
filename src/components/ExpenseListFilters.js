import React from 'react';
import { connect } from 'react-redux';
import 'react-dates/initialize';
import { DateRangePicker } from 'react-dates';
import { setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate } from '../actions/filters';

export class ExpenseListFilters extends React.Component {
    state = {
        calenderFocused: null,

    }
    onDateChange = ({ startDate, endDate }) => {
        this.props.setStartDate(startDate);
        this.props.setEndDate(endDate);
    }
    onFocusChange = (calenderFocused) => {
        this.setState(() => ({ calenderFocused }))

    }
    onTextChange = (e) => {
        this.props.setTextFilter(e.target.value);
    }
    onSortChange = (e) => {
        if (e.target.value === "date") {
            this.props.sortByDate();

        } else if (e.target.value === "amount") {
            this.props.sortByAmount();
        }
    }
    render() {
        return (
            <div>
                <input type='text' value={this.props.filters.text}
                    onChange={this.onTextChange} />
                <select
                    value={this.props.filters.sortBy}
                    onChange={this.onSortChange}

                >
                    <option value="date">Date</option>
                    <option value="amount">Amount</option>
                </select>
                <DateRangePicker
                    startDate={this.props.filters.startDate}
                    startDateId="0"
                    endDate={this.props.filters.endDate}
                    endDateId="1"
                    onDateChange={this.onDateChange}
                    focusedInput={this.state.calenderFocused}
                    onFocusChange={this.onFocusChange}
                    numberOfMonths={1}
                    isOutsideRange={() => false}
                    showClearDates={true}


                />

            </div>


        )
    }
}

const mapStateToProps = (state) => 
   (  {
        filters: state.filters
    })


const mapDispatchToProps = (dispatch) => ({
    setTextFilter: (text)=> dispatch(setTextFilter(text)),
    sortByDate:()=> dispatch(sortByDate()),
    sortByAmount: ()=> dispatch(sortByAmount()),
    setStartDate: (startDate)=> dispatch(setStartDate(startDate)),
    setEndDate: (endDate)=> dispatch(setEndDate(endDate))
}
)
export default connect(mapStateToProps,mapDispatchToProps)(ExpenseListFilters);