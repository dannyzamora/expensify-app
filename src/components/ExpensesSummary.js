import React from 'react';
import { connect } from 'react-redux';
import {Link} from 'react-router-dom';
import numeral from 'numeral';

import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';


export const ExpenseSummary = ({ expensesTotal, expensesCount }) => {
    const expenseWord = expensesCount === 1 ? 'expense' : 'expenses';
    const formattedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00');
    return (
        <div className="page-header">
            <div className="content-container">
                <h1 className = "page-header__title">Viewing <span>{expensesCount}</span> {expenseWord} totalling <span>{formattedExpensesTotal}</span></h1>
                <div className= "page-header__actions">
                    <Link className= "button" to ="/create">Add Expense</Link>
                </div>
            </div>
        </div>
    )
}




const mapStatetoProps = (state) => {
    const curexpenses = selectExpenses(state.expenses, state.filters);

    return {
        expensesTotal: selectExpensesTotal(curexpenses),
        expensesCount: curexpenses.length
    }
}

export default connect(mapStatetoProps)(ExpenseSummary);
// export class ExpenseSummary extends React.Component {

//     render() {
//         <div>
//             {
//                 numeral(amount/100).format('$0,0.00')
//             }



//         </div>
//     }
// }

// const mapStatetoProps = (state) => {
//     return {
//         expenses: selectExpenses(state.expenses, state.filters)
//     }
// }
// const mapDispatchToProps = (dispatch) => ({
//     selectExpensesTotal:(expenses) => dispatch(selectExpensesTotal(expenses))
//   });

// export default connect(mapStatetoProps,mapDispatchToProps)(ExpenseList);