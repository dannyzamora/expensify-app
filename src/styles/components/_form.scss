.form {
  display: flex;
  flex-direction: column;
  >* {
    margin-bottom: $m-size;
    .SingleDatePickerInput {
      width: 100%;
  }
  }
}

.form__error {
  margin: 0 0 $m-size 0;
  font-style: italic;
}
