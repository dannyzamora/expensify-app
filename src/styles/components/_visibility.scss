.show-for-mobile{
    @media (min-width: $desktop-breakpoint) {
        display: none;
      }
}

.short-for-desktop{
    @media (max-width: $desktop-breakpoint - .01rem) {
        display: none;
      }
}