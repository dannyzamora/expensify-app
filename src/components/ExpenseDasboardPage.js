import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';

const ExpenseDasboardPage=()=> (
    <div>
        <ExpenseListFilters/>
        <ExpenseList/>
        
    </div>
)
export default ExpenseDasboardPage;