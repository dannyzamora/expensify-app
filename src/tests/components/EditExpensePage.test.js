import React from "react";
import { shallow } from 'enzyme';
import { EditExpensePage } from '../../components/EditExpensePage';
import expenses from '../fixtures/expenses';

let startEditExpenses, startRemoveExpenses, history, wrapper;

beforeEach(() => {
    startEditExpenses = jest.fn();
    startRemoveExpenses = jest.fn();
    history = { push: jest.fn() };
    wrapper = shallow(
        <EditExpensePage
            startEditExpenses={startEditExpenses}
            startRemoveExpenses={startRemoveExpenses}
            history={history} 
            expense= {expenses[2]}/>
                       
)
});

test("Should render editexpensepage",()=>{
    expect(wrapper).toMatchSnapshot();
});

test("should handle startEditExpenses",()=>{
    wrapper.find('ExpenseForm').prop('onSubmit')(expenses[2]);
    expect(history.push).toHaveBeenLastCalledWith("/");
    expect(startEditExpenses).toHaveBeenLastCalledWith(expenses[2].id,expenses[2])

});

test("should handle startRemoveExpenses",()=>{
    wrapper.find('button').simulate('click');
    expect(history.push).toHaveBeenLastCalledWith("/");
    expect(startRemoveExpenses).toHaveBeenLastCalledWith({
        id:expenses[2].id
    })

});