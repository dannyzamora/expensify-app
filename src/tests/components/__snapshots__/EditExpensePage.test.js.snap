// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render editexpensepage 1`] = `
<div>
  <ExpenseForm
    expense={
      Object {
        "amount": 4500,
        "createdAt": 345600000,
        "description": "CREDITCARD",
        "id": "3",
        "note": "",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
